cernopendata/config.py

```diff
@@ -341,42 +350,43 @@
             signature=dict(terms=dict(
                 field='signature.keyword',
                 order=dict(_term='asc'))),
-            event_number={
-                'range': {
-                    'field': 'distribution.number_events',
-                    'ranges': [
-                        {
-                            'key': '0--999',
-                            'from': 0,
-                            'to': 999
-                        },
-                        {
-                            'key': '1000--9999',
-                            'from': 1000,
-                            'to': 9999
-                        },
-                        {
-                            'key': '10000--99999',
-                            'from': 10000,
-                            'to': 99999
-                        },
-                        {
-                            'key': '100000--999999',
-                            'from': 100000,
-                            'to': 999999
-                        },
-                        {
-                            'key': '1000000--9999999',
-                            'from': 1000000,
-                            'to': 9999999
-                        },
-                        {
-                            'key': '10000000--',
-                            'from': 10000000
-                        }
-                    ]
-                }
-            }
+            # TODO: Breaks the search (KeyError: 'query'), adapt or remove?
```

Comment:
@ParthS007: This is one the issues I had. Do you have a clue how to adapt it? cc/ @tiborsimko @diegodelemos

Reply:
Since the range filter hasn't changed since the last versions, I think the problem is merely presentational which is good. Invenio-Search-UI does not return config to React-SearchKit other than for [term aggregations](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-terms-aggregation.html) (see [here](https://github.com/inveniosoftware/invenio-search-ui/blob/44b6122d8f066a92344f1c1a58b4d18e888b690c/invenio_search_ui/views.py#L191-L211)), but we also need [range aggregations](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-range-aggregation.html).

From what I understand the fix needs to be done in Invenio-Search-UI, allowing to return config for other kinds of aggregations. A _dirty local fix_ to see if things work now is to modify `/opt/invenio/var/instance/python/lib/python3.6/site-packages/invenio_search_ui/views.py` as follows:
```diff
        return [
            {
                "title": k.capitalize(),
                "agg": {
                    "aggName": k,
-                   "field": v["terms"]["field"]
+                   "field": (v.get("terms") or v.get("range")).get("field")
                },
            }
            for k, v in aggs.items()

```
Note: Of course this is not a good solution, I guess we need a function that gets `field` for any of the supported aggregations? What do you think?

Reply:
Thanks for the detailed explanation, now I understand better the issue. I've tested your snippet and it stopped failing.
I guess there's not much we can do apart from contributing upstream to `invenio-search-ui`? How come that it was working well with the old version of `invenio-search-ui`?

Reply:
OK, I see that's being used [here](https://github.com/inveniosoftware/invenio-search-ui/blob/44b6122d8f066a92344f1c1a58b4d18e888b690c/invenio_search_ui/assets/semantic-ui/js/invenio_search_ui/components/SearchApp.js#L37) to display the multiple bucket aggregation components.
![image](https://user-images.githubusercontent.com/8863238/105751569-7928d280-5f46-11eb-9d3b-ef72c448e57a.png)

It shouldn't be a big deal to sync with Invenio team to contribute upstream, should it?

Reply:
> It shouldn't be a big deal to sync with Invenio team to contribute upstream, should it? 

No big deal, I just wanted to first double-check with you. Since you can reproduce it and it seems like a reasonable request we can ask and contribute to extend this behavior.

> How come that it was working well with the old version of `invenio-search-ui`?

Probably because all this info was populated [through Jinja templates](https://github.com/inveniosoftware/invenio-search-ui/blob/master/invenio_search_ui/templates/invenio_search_ui/search.html), and now there is this new function which is [specific to React SearchKit](https://github.com/inveniosoftware/invenio-search-ui/blob/44b6122d8f066a92344f1c1a58b4d18e888b690c/invenio_search_ui/views.py#L194) to render the facets through an API call.